code/model/modelanimation.cpp

```diff
@@ -894,65 +753,112 @@ namespace animation {
 		}
+		else
+			return getSubmodel(other->m_name);
+	}
+
+
+	void anim_set_initial_states(ship* shipp) {
+		ship_info* sip = &Ship_info[shipp->ship_info_index];
+		sip->animations.clearShipData(model_get_instance(shipp->model_instance_num));
+		sip->animations.startAll(model_get_instance(shipp->model_instance_num), animation::ModelAnimationTriggerType::Initial, ModelAnimationDirection::FWD, true, true);
+		sip->animations.startAll(model_get_instance(shipp->model_instance_num), animation::ModelAnimationTriggerType::OnSpawn, ModelAnimationDirection::FWD);
 	}
 
-	const std::map<ModelAnimationTriggerType, const char*> Animation_type_names = {
-	{ModelAnimationTriggerType::Initial, "initial"},
-	{ModelAnimationTriggerType::Docking_Stage1, "docking-stage-1"},
-	{ModelAnimationTriggerType::Docking_Stage2, "docking-stage-2"},
-	{ModelAnimationTriggerType::Docking_Stage3, "docking-stage-3"},
-	{ModelAnimationTriggerType::Docked, "docked"},
-	{ModelAnimationTriggerType::PrimaryBank, "primary-bank"},
-	{ModelAnimationTriggerType::SecondaryBank, "secondary-bank"},
-	{ModelAnimationTriggerType::DockBayDoor, "fighterbay"},
-	{ModelAnimationTriggerType::Afterburner, "afterburner"},
-	{ModelAnimationTriggerType::TurretFiring, "turret-firing"},
-	{ModelAnimationTriggerType::Scripted, "scripted"},
-	{ModelAnimationTriggerType::TurretFired, "turret-fired"}
+	const std::map<ModelAnimationTriggerType, std::pair<const char*, bool>> Animation_types = {
+	{ModelAnimationTriggerType::Initial, {"initial", false}}, //Atypical case. Will only ever be run in fully triggered state and then applied as base transformation
+	{ModelAnimationTriggerType::OnSpawn, {"on-spawn", false}}, //Atypical case. While no reverse trigger occurs, it is also guaranteed to not trigger more than once per model, hence non-resetting animations are fine here.
+	{ModelAnimationTriggerType::Docking_Stage1, {"docking-stage-1", false}},
+	{ModelAnimationTriggerType::Docking_Stage2, {"docking-stage-2", false}},
+	{ModelAnimationTriggerType::Docking_Stage3, {"docking-stage-3", false}},
+	{ModelAnimationTriggerType::Docked, {"docked", false}},
+	{ModelAnimationTriggerType::PrimaryBank, {"primary-bank", false}},
+	{ModelAnimationTriggerType::SecondaryBank, {"secondary-bank", false}},
+	{ModelAnimationTriggerType::DockBayDoor, {"fighterbay", false}},
+	{ModelAnimationTriggerType::Afterburner, {"afterburner", false}},
+	{ModelAnimationTriggerType::TurretFiring, {"turret-firing", false}},
+	{ModelAnimationTriggerType::Scripted, {"scripted", false}},
+	{ModelAnimationTriggerType::TurretFired, {"turret-fired", true}},
+	{ModelAnimationTriggerType::PrimaryFired,   {"primary-fired", true}},
+	{ModelAnimationTriggerType::SecondaryFired, {"secondary-fired", true}},
+	{ModelAnimationTriggerType::WeaponLaunched, {"weapon-launched", false}} //Atypical case. While no reverse trigger occurs, it is also guaranteed to not trigger more than once per model, hence non-resetting animations are fine here.
```

Comment:
Isn't this the same as `on-spawn`, but for weapons rather than ships?

Reply:
Yes, it is.
I'll change on-spawn to replace weapon-launched